src/jit-tp-analyze/Program.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+
+namespace ManagedCodeGen;
+
+internal partial class Program
+{
+    private static readonly Regex _traceLineRegex = new("(\\d+) +: (.*)", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+    private readonly string _baseTracePath;
+    private readonly string _diffTracePath;
+    private readonly double _noise;
+
+    public Program(JitTpAnalyzeRootCommand command, ParseResult result)
+    {
+        _baseTracePath = result.GetValue(command.BasePath);
+        _diffTracePath = result.GetValue(command.DiffPath);
+        _noise = result.GetValue(command.Noise);
+    }
+
+    public void Run()
+    {
+        TextWriter output = Console.Out;
+        Dictionary<string, long> baseTrace = ParseTrace(_baseTracePath);
+        Dictionary<string, long> diffTrace = ParseTrace(_diffTracePath);
+        HashSet<string> allRecordedFunctions = new();
+        foreach (var function in baseTrace)
+        {
+            allRecordedFunctions.Add(function.Key);
+        }
+        foreach (var function in diffTrace)
+        {
+            allRecordedFunctions.Add(function.Key);
+        }
+
+        long baseTotalInsCount = baseTrace.Sum(x => x.Value);
+        long diffTotalInsCount = diffTrace.Sum(x => x.Value);
+        double totalPercentageDiff = GetPercentageDiff(baseTotalInsCount, diffTotalInsCount);
+        output.WriteLine($"Base: {baseTotalInsCount}, Diff: {diffTotalInsCount}, {FormatPercentageDiff(totalPercentageDiff, "0000")}");
+        output.WriteLine();
+
+        // Now create a list of functions which contributed to the difference.
+        long totalAbsInsCountDiff = 0;
+        List<FunctionDiff> diffs = new();
+        foreach (string functionName in allRecordedFunctions)
+        {
+            long diffInsCount = diffTrace.GetValueOrDefault(functionName);
+            long baseInsCount = baseTrace.GetValueOrDefault(functionName);
+            long insCountDiff = diffInsCount - baseInsCount;
+            if (insCountDiff is 0)
```

Comment:
Does this work? What's the difference between this and:
```suggestion
            if (insCountDiff == 0)
```

Reply:
It does, this is the same pattern matching as `x is null`. There is no difference for primitives. I've changed it back to `==`.